src/Compilers/CSharp/Test/Emit3/OverloadResolutionPriorityTests.cs

```diff
@@ -0,0 +1,1768 @@
+                void I.M(string s) => throw null;
+
+                [OverloadResolutionPriority(1)]
+                int I.this[object o]
+                {
+                    get => throw null;
+                    set => throw null;
+                }
+                int I.this[string s]
+                {
+                    get => throw null;
+                    set => throw null;
+                }
+            }
+            """;
+
+        CreateCompilation([source, OverloadResolutionPriorityAttributeDefinition]).VerifyDiagnostics(
+            // (15,6): error CS9503: Cannot use 'OverloadResolutionPriorityAttribute' on an explicit interface implementation.
+            //     [OverloadResolutionPriority(1)]
+            Diagnostic(ErrorCode.ERR_CannotApplyOverloadResolutionPriorityToExplicitImplementation, "OverloadResolutionPriority(1)").WithLocation(15, 6),
+            // (19,6): error CS9503: Cannot use 'OverloadResolutionPriorityAttribute' on an explicit interface implementation.
+            //     [OverloadResolutionPriority(1)]
+            Diagnostic(ErrorCode.ERR_CannotApplyOverloadResolutionPriorityToExplicitImplementation, "OverloadResolutionPriority(1)").WithLocation(19, 6)
+        );
+    }
+
+    [Theory, CombinatorialData]
+    public void Dynamic(bool useMetadataReference)
+    {
+        var source = @"
+            using System.Runtime.CompilerServices;
+
+            public class C
+            {
+                [OverloadResolutionPriority(1)]
+                public C(object o) => throw null;
+                public C(string s) => System.Console.Write(1);
+
+                [OverloadResolutionPriority(1)]
+                public void M(object o) => throw null;
+                public void M(string s) => System.Console.Write(2);
+
+                [OverloadResolutionPriority(1)]
+                public int this[object o]
+                {
+                    get => throw null;
+                    set => throw null;
+                }
+                public int this[string o]
+                {
+                    get { System.Console.Write(3); return 1; }
+                    set => System.Console.Write(4);
+                }
+            }
+            ";
+
+        var executable = """
+            dynamic arg = "test";
+            C c = new C(arg);
+            c.M(arg);
+            _ = c[arg];
+            c[arg] = 1;
+            """;
+
+        CompileAndVerify([executable, source, OverloadResolutionPriorityAttributeDefinition], targetFramework: TargetFramework.Mscorlib45AndCSharp, expectedOutput: "1234").VerifyDiagnostics();
+
+        var comp = CreateCompilation([source, OverloadResolutionPriorityAttributeDefinition], targetFramework: TargetFramework.Mscorlib45AndCSharp);
+        CompileAndVerify(executable, references: new[] { AsReference(comp, useMetadataReference) }, targetFramework: TargetFramework.Mscorlib45AndCSharp, expectedOutput: "1234").VerifyDiagnostics();
+    }
```

Comment:
1. I can add tests for binary operators.
2. I can add tests for method groups.
3. `LangVersion`
4. They all go through overload resolution. While I can add tests for some of these, I'm not sure how interesting they actually are, since it's just a call to the `OverloadResolution` type.